site/src/pages/TerminalPage/TerminalAlerts.tsx

```diff
@@ -0,0 +1,121 @@
+      dismissible
+      actions={<RefreshSessionButton />}
+    >
+      The workspace{" "}
+      <Link
+        title="startup script has exited with an error"
+        href={docs("/templates#startup-script-exited-with-an-error")}
+        target="_blank"
+        rel="noreferrer"
+      >
+        startup script has exited with an error
+      </Link>
+      , we recommend reloading this session and{" "}
+      <Link
+        title=" debugging the startup script"
+        href={docs("/templates#debugging-the-startup-script")}
+        target="_blank"
+        rel="noreferrer"
+      >
+        debugging the startup script
+      </Link>{" "}
+      because{" "}
+      <Link
+        title="your workspace may be incomplete."
+        href={docs("/templates#your-workspace-may-be-incomplete")}
+        target="_blank"
+        rel="noreferrer"
+      >
+        your workspace may be incomplete.
+      </Link>{" "}
+    </TerminalAlert>
+  );
+};
+
+export const LoadingScriptsAlert = () => {
+  return (
+    <TerminalAlert
+      dismissible
+      severity="info"
+      actions={<RefreshSessionButton />}
+    >
+      Startup scripts are still running. You can continue using this terminal,
+      but{" "}
+      <Link
+        title="your workspace may be incomplete."
+        href={docs("/templates#your-workspace-may-be-incomplete")}
+        target="_blank"
+        rel="noreferrer"
+      >
+        {" "}
+        your workspace may be incomplete.
+      </Link>
+    </TerminalAlert>
+  );
+};
+
+export const LoadedScriptsAlert = () => {
+  return (
+    <TerminalAlert
+      severity="success"
+      dismissible
+      actions={<RefreshSessionButton />}
+    >
+      Startup scripts have completed successfully. The workspace is ready but
+      this{" "}
+      <Link
+        title="session was started before the startup scripts finished"
+        href={docs("/templates#your-workspace-may-be-incomplete")}
+        target="_blank"
+        rel="noreferrer"
+      >
+        session was started before the startup script finished.
+      </Link>{" "}
+      To ensure your shell environment is up-to-date, we recommend reloading
+      this session.
+    </TerminalAlert>
+  );
+};
+
+const TerminalAlert = (props: AlertProps) => {
+  return (
+    <Alert
+      {...props}
+      sx={{
+        borderRadius: 0,
+        borderWidth: 0,
+        borderBottomWidth: 1,
+        borderBottomColor: (theme) => theme.palette.divider,
+        backgroundColor: (theme) => theme.palette.background.paperLight,
+        borderLeft: (theme) =>
+          `3px solid ${theme.palette[props.severity!].light}`,
+        marginBottom: 1,
+      }}
+    />
+  );
+};
+
+export const DisconnectedAlert = (props: AlertProps) => {
+  return (
+    <TerminalAlert {...props} severity="warning">
+      Disconnected
+    </TerminalAlert>
+  );
+};
+
+const RefreshSessionButton = () => {
+  return (
+    <Button size="small" variant="text" onClick={window.location.reload}>
```

Comment:
This button is not working for me (but the other reload button works), my guess is because of how `this` is being bound.  Either one of these works:

```
onClick={() => window.location.reload()}>
```

```
onClick={window.location.reload.bind(window.location)}>
```

Or maybe we could roll this into a single reload function that is shared with the other reload button so we get the same text change and disable button effect.

Reply:
Ahhh ok, going to fix it right now.